sensor/kubernetes/eventpipeline/output/output_impl.go

```diff
@@ -18,7 +18,8 @@ type outputQueueImpl struct {
 	stopSig      concurrency.Signal
 }
 
-// Send a ResourceEvent message to the inner queue
+// Send a ResourceEvent to outside the pipeline. This will trigger alert detection if component.ResourceEvent
```

Comment:
`outside the pipeline`? Which pipeline, this is new to me? :thinking: :D

Reply:
When the only responsibility of the Output Pipeline sending events to Detector is, could it be the `DetectorQueue`?
When it does more and can dispatch ResourceEvents, could it be that the output queue is used to register components on it which want to perform actions on Kubernetes events?

Reply:
Ah, below it is documented, I think linking the data structure to reference the pipeline directly here is beneficial. When looking at the source later I would not find the Pipeline directly to read up on the documentation there.

Reply:
Oh, I see that there is probably a confusion in the two types of "components". There are sensor components and pipeline components.

- Sensor components is a design that has been there for a long time. They have an interface that allows every component to send messages to central by returning a queue in the `ResponsesC` function.
- The "Event Pipeline" that we are discussing here **is a sensor component**. I.e. it can send messages to the gRPC stream.
- Pipeline components are these sub-components inside the event pipeline that process a kubernetes event from start to finish (listener, resolver and output are pipeline components) 

```
[Event Pipeline (Sensor Component)] -> ResponsesC -> gRPC Stream
 |                       
 | <contains>
 V                         
 [Listener -> Resolver -> Output] (Pipeline components)
                                |
                                | <sends messages to>
                                V
                [Detector (Sensor component)] -> ResponsesC -> gRPC stream

```

The event pipeline will take whatever output is writing in its channel and copy into the ResponseC channel that is polled by the gRPC stream. 

The output component will *also* send data to detector, which is *another sensor component* and by itself it is going to write into its own ResponsesC channel. 

There was a plan in the beginning of re-sync to move detector *entirely* into the event pipeline. Maybe this would've reduced the confusion. But it was a major refactor, and we decided against it for simplicity.

Reply:
Thanks for the good explanation Fred, this is great!